source/ncconvert/converters/GeometryConverter.cpp

```diff
@@ -48,17 +48,37 @@ auto ReadFbx(const std::filesystem::path& path, Assimp::Importer* importer, unsi
     {
         throw nc::NcError("Fbx contains no mesh data\n    file: ", path.string());
     }
-    else if (scene->mNumMeshes > 1)
+
+    return scene;
+}
+
+auto GetMeshFromScene(const aiScene* scene, const std::optional<std::string>& subResourceName = std::nullopt) -> aiMesh*
+{
+    aiMesh* mesh = nullptr;
```

Comment:
IMO this function would read a little easier if you swapped the cases - first check if no subresource -> return first mesh. 

```cpp
NC_ASSERT(scene->mNumMeshes, "...");
if (!subResourcename.has_value())
{
   return scene->mMeshes[0];
}

// Other code, but slightly less indented
```

Reply:
Dealers choice between assert and the if/throw you have. NC_ASSERT is on (at least here) even in release builds.